### PR TITLE
Beginner tutorial for train resnet on HSN

### PR DESCRIPTION
**Pull request by PariaValizadeh**
train resnet on Hsn to close [issue](https://www.notion.so/Get-familiar-with-BirdSet-pipeline-41b59a64fe11407b8b102bad0d86f767?pvs=4)

## Roadmap
- [x] get datamodule
- [x] get one sample
- [x] visualize spetrogram
- [x] listen to audio
- [x] get trainer
- [x] initialize model
- [x] train model
- [x] eval model
- [x] inference on on sample
- [x] get ebirdcode from one sample

### CODE REVIEW

**Review by raphaelschwinger — CHANGES_REQUESTED** [review]
Hey @PariaValizadeh,
Thanks for your great work and sorry it took me so long to review your PR.
Really good job for your first PR 🙌 !
Here are a few things you can improve before we can merge it into main:
- Clarify the name of the PR as this will be the commit headline it is important to be precise here
- Add a description of what the PR is doing
- Remove the other notebooks from this PR as this one is just about extending the beginner tutorial. Usually you are doing one PR for each new feature/thing you are doing to improve the readability of the git history and make it easy to redo changes
- Move the "Mapping of Labels to eBird Codes" down to your model training section to avoid duplicates
- Improve the training of the network to have some reasonable results e.g. 20% cmAP (playaround with longer training or try out an efficientnet instead of a resnet). If you think it is not feasible to train a network within the notebook add description that this is just a test for the pipeline.
- Add the ground truth label from the dataset for comparison

File: notebooks/tutorials/birdset-pipeline_tutorial.ipynb [hunk 1]
```diff
@@ -2165,6 +2165,473 @@
...
+            "cell_type": "code",
+            "execution_count": null,
+            "id": "b50dc342",
+            "metadata": {},
+            "outputs": [],
+            "source": [
+                "#from train_loader = dm.train_dataloader() import Trainer\n",
+                "from lightning import Trainer\n",
+                "min_epochs = 1\n",
+                "max_epochs = 5\n",
```
[the remark this answers is not in the document]
Comment (PariaValizadeh) [reply notebooks/tutorials/birdset-pipeline_tutorial.ipynb:2255]: [7a6cb3984ef8fe70400442af78ea0a7f7e32d04c](url)
I have the epoch number and selectiong gpu part

[the remark this answers is not in the document]
Comment (PariaValizadeh) [reply notebooks/tutorials/birdset-pipeline_tutorial.ipynb:2255]: I resolve it and 
[7a6cb3984ef8fe70400442af78ea0a7f7e32d04c](url)
also add some description[49469e59dd11775695dec93542410c84e5398c9f](url)

File: notebooks/tutorials/birdset-pipeline_tutorial.ipynb [hunk 2]
```diff
@@ -2165,6 +2165,473 @@
...
+                "trainer = Trainer(min_epochs=min_epochs, max_epochs=max_epochs, accelerator=\"gpu\", devices=[1])"
```
[the remark this answers is not in the document]
Comment (PariaValizadeh) [reply notebooks/tutorials/birdset-pipeline_tutorial.ipynb:2256]: changed it 7a6cb3984ef8fe70400442af78ea0a7f7e32d04c

File: notebooks/tutorials/birdset-pipeline_tutorial.ipynb [hunk 3]
```diff
@@ -2165,6 +2165,473 @@
...
+                "print(batch[\"labels\"].shape)\n",
+                "batch"
+            ]
+        },
+        {
+            "cell_type": "markdown",
+            "id": "9f733052",
+            "metadata": {},
+            "source": [
+                "**Step2: Training the Model**"
```
[the remark this answers is not in the document]
Comment (PariaValizadeh) [reply notebooks/tutorials/birdset-pipeline_tutorial.ipynb:2242]: Done

File: notebooks/tutorials/birdset-pipeline_tutorial.ipynb [hunk 4]
```diff
@@ -2165,6 +2165,473 @@
                 "\n",
                 "    return mapping"
             ]
+        },
+        {
+            "cell_type": "markdown",
+            "id": "b1993d85",
+            "metadata": {},
+            "source": [
+                "**Train and evaluate a Model**"
```
[the remark this answers is not in the document]
Comment (PariaValizadeh) [reply notebooks/tutorials/birdset-pipeline_tutorial.ipynb:2174]: Done

File: notebooks/tutorials/birdset-pipeline_tutorial.ipynb [hunk 5]
```diff
@@ -2165,6 +2165,473 @@
...
+            "metadata": {},
+            "outputs": [],
+            "source": [
+                "#from train_loader = dm.train_dataloader() import Trainer\n",
+                "from lightning import Trainer\n",
+                "min_epochs = 1\n",
+                "max_epochs = 5\n",
+                "trainer = Trainer(min_epochs=min_epochs, max_epochs=max_epochs, accelerator=\"gpu\", devices=[1])"
+            ]
+        },
```
[the remark this answers is not in the document]
Comment (PariaValizadeh) [reply notebooks/tutorials/birdset-pipeline_tutorial.ipynb:2258]: because i define max epochs in it I think the current position might be better but there is also possible to define maxepochs twice

File: notebooks/tutorials/birdset-pipeline_tutorial.ipynb [hunk 6]
```diff
@@ -2165,6 +2165,473 @@
...
+                "transforms = BirdSetTransformsWrapper(model_type='vision',preprocessing=PreprocessingConfig(spectrogram_conversion= Spectrogram(\n",
+                "            n_fft=1024,\n",
+                "            hop_length=320,\n",
+                "            power=2.0,\n",
+                "        ),), decoding=EventDecoding(sampling_rate=32000), task=\"multilabel\")\n",
+                "\n",
+                "# initiate the data module\n",
+                "dm = BirdSetDataModule(\n",
+                "    dataset= DatasetConfig(\n",
+                "        data_dir='/workspace/data_birdset/HSN',\n",
```
[the remark this answers is not in the document]
Comment (PariaValizadeh) [reply notebooks/tutorials/birdset-pipeline_tutorial.ipynb:2208]: done 39a7b27da22f2da22d2919b6acf0830327773bbf

File: notebooks/tutorials/beginner_tutorials/train_Resnet_on_HSN.ipynb [hunk 7]
```diff
@@ -0,0 +1,1131 @@
...
+    "\n",
+    "transforms = BirdSetTransformsWrapper(model_type='vision',preprocessing=PreprocessingConfig(spectrogram_conversion= Spectrogram(\n",
+    "            n_fft=1024,\n",
+    "            hop_length=320,\n",
+    "            power=2.0,\n",
+    "        ),), decoding=EventDecoding(sampling_rate=32000), task=\"multilabel\")\n",
+    "# initiate the data module\n",
+    "dm = BirdSetDataModule(\n",
+    "    dataset= DatasetConfig(\n",
+    "        data_dir='../../data_birdset/HSN',\n",
```
Comment (raphaelschwinger) [inline notebooks/tutorials/beginner_tutorials/train_Resnet_on_HSN.ipynb:296]:
```suggestion
    "        data_dir='../../../data_birdset/HSN',\n",
```

File: notebooks/tutorials/beginner_tutorials/train_Resnet_on_HSN.ipynb [hunk 8]
```diff
@@ -0,0 +1,1131 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from lightning import Trainer \n",
+    "\n",
+    "min_epochs = 1\n",
+    "max_epochs = 20\n",
+    "trainer = Trainer(min_epochs=min_epochs, max_epochs=max_epochs, accelerator=\"gpu\", devices=0)"
```
Comment (raphaelschwinger) [inline notebooks/tutorials/beginner_tutorials/train_Resnet_on_HSN.ipynb:359]:
```suggestion
    "trainer = Trainer(min_epochs=min_epochs, max_epochs=max_epochs, accelerator=\"gpu\", devices=[0])"
```
This must have thrown an error. Always run your code before commiting ;)

File: notebooks/tutorials/birdset-pipeline_tutorial.ipynb
Comment (raphaelschwinger) [inline notebooks/tutorials/birdset-pipeline_tutorial.ipynb:1]: @PariaValizadeh Could you remove all changes on this file as we agreed to move the training tutorial into a sperate file. Thanks!

File: notebooks/tutorials/beginner_tutorials/audiovisualization.ipynb
Comment (raphaelschwinger) [inline notebooks/tutorials/beginner_tutorials/audiovisualization.ipynb:1]: @PariaValizadeh Could you add documentation?

**Review by raphaelschwinger — CHANGES_REQUESTED** [review]
Hey @PariaValizadeh I gave your PR another review ;) Thanks for your changes!

File: notebooks/tutorials/beginner_tutorials/train_Resnet_on_HSN.ipynb (hunk 7 above)
Comment (PariaValizadeh) [reply notebooks/tutorials/beginner_tutorials/train_Resnet_on_HSN.ipynb]: done 21b98f0a88efbc3df5cb9d0b5284fa4c1c7453dd

File: notebooks/tutorials/beginner_tutorials/train_Resnet_on_HSN.ipynb (hunk 8 above)
Comment (PariaValizadeh) [reply notebooks/tutorials/beginner_tutorials/train_Resnet_on_HSN.ipynb]: done a70be7bddfe5018a8edd173a0bb6a6948328e9dc

File: notebooks/tutorials/birdset-pipeline_tutorial.ipynb
Comment (PariaValizadeh) [reply notebooks/tutorials/birdset-pipeline_tutorial.ipynb:1]: okay sure

File: notebooks/tutorials/beginner_tutorials/audiovisualization.ipynb
Comment (PariaValizadeh) [reply notebooks/tutorials/beginner_tutorials/audiovisualization.ipynb:1]: I just make a pull reguest with documentation 69c91ff263f6068814db38f07c0be1982bf57b23

Comment (PariaValizadeh) [reply notebooks/tutorials/beginner_tutorials/audiovisualization.ipynb:1]: I tried to clean up the pr 04cf1e1d62d85d86642cb99316539eb4c54cd3af

**Comment by lurauch** [comment]
@PariaValizadeh Same as the other two comments: spelling & naming conventions

**Comment by PariaValizadeh** [comment]
> @PariaValizadeh Same as the other two comments: merge conflict, spelling & naming conventions

Thank you. I fixed it 515960e975ca7f36eca27d52684e4ad36888d9b1
